File: 2485. Find the Pivot Integer/pivotInteger.js
/**
 * @param {number} n
 * @return {number}
 */
const pivotInteger = (n) => {
  /*
      1 + 2 + ... + x = x + ... + n
    ((1 + x) * x) / 2 = ((x + n) * (n - x + 1)) / 2
            x + x ^ 2 = x * n - x ^ 2 + x + n ^ 2 - n * x + n
            2 * x ^ 2 = n ^ 2 + n
            x = sqrt((n ^ 2 + n) / 2)
  */
  const x = Math.floor((n * n + n) / 2);
  const y = Math.floor(Math.sqrt(y));

  return y * y === x ? y : -1;
};

console.log('pivotInteger: ', pivotInteger(8))

File: 2485. Find the Pivot Integer/pivotInteger.py
import math

class Solution:
  def pivotInteger(self, n: int) -> int:
    x = (n ** 2 + n) // 2
    y = int(math.sqrt(x))

    return x if x * x == y else -1
  
solution = Solution()
print('pivotInteger: ', solution.pivotInteger(8))
